chainerrl/misc/pretrained_models.py

```diff
@@ -0,0 +1,149 @@
+"""This file is a fork from ChainerCV, an MIT-licensed project,
+https://github.com/chainer/chainercv/blob/master/chainercv/utils/download.py
+"""
+
+
+from __future__ import print_function
+from __future__ import division
+from __future__ import unicode_literals
+from __future__ import absolute_import
+from builtins import *  # NOQA
+from future import standard_library
+standard_library.install_aliases()  # NOQA
+
+from distutils.util import strtobool
+import filelock
+import hashlib
+import os
+import shutil
+import tempfile
+import time
+import sys
+import zipfile
+from six.moves.urllib import request
+
+from chainer.dataset.download import get_dataset_directory
+from chainer.dataset.download import get_dataset_root
+
+
+MODELS = {
+    "DQN": ["best", "final"],
+    "IQN": ["best", "final"],
+    "Rainbow": ["best", "final"],
+    "A3C": ["final"],
+    "DDPG": ["best", "final"],
+    "TRPO": ["best", "final"],
+    "PPO": ["final"],
+    "TD3": ["best", "final"],
+    "SAC": ["best", "final"]
+}
+
+url = "https://chainer-assets.preferred.jp/chainerrl/"
+
+
+def _reporthook(count, block_size, total_size):
+    global start_time
+    if count == 0:
+        start_time = time.time()
+        print('  %   Total    Recv       Speed  Time left')
+        return
+    duration = time.time() - start_time
+    progress_size = count * block_size
+    try:
+        speed = progress_size / duration
+    except ZeroDivisionError:
+        speed = float('inf')
+    percent = progress_size / total_size * 100
+    eta = int((total_size - progress_size) / speed)
+    sys.stdout.write(
+        '\r{:3.0f} {:4.0f}MiB {:4.0f}MiB {:6.0f}KiB/s {:4d}:{:02d}:{:02d}'
+        .format(
+            percent, total_size / (1 << 20), progress_size / (1 << 20),
+            speed / (1 << 10), eta // 60 // 60, (eta // 60) % 60, eta % 60))
+    sys.stdout.flush()
+
+
+def cached_download(url):
+    """Downloads a file and caches it.
+    This is different from the original
+    :func:`~chainer.dataset.cached_download` in that the download
+    progress is reported. Note that this progress report can be disabled
+    by setting the environment variable `CHAINERCV_DOWNLOAD_REPORT` to `'OFF'`.
+    It downloads a file from the URL if there is no corresponding cache. After
+    the download, this function stores a cache to the directory under the
+    dataset root (see :func:`set_dataset_root`). If there is already a cache
+    for the given URL, it just returns the path to the cache without
+    downloading the same file.
+    Args:
+        url (string): URL to download from.
+    Returns:
+        string: Path to the downloaded file.
+    """
```

Comment:
- Why did you remove the empty lines in the docstring? We use Google Style docstrings http://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings
- It seems like you added some change to this function, e.g., 'CHAINERRL_DOWNLOAD_REPORT'. Can you change the docstring accordingly and highlight the changes from ChainerCV's function?

Reply:
Addressed. I removed the `CHAINERRL_DOWNLOAD_REPORT`. I had previously walked through the ChainerCV docstring and replaced references to ChainerCV with ChainerRL. The `CHAINERRL_DOWNLOAD_REPORT ` is not relevant to CRL.

Reply:
> The `CHAINERRL_DOWNLOAD_REPORT` is not relevant to CRL.

This function's behavior depends on `CHAINERRL_DOWNLOAD_REPORT`, so you should either document it or remove the dependency.

Reply:
I removed the dependency.